File: solutions/java/MigratoryBird.java
public class MigratoryBird{
	private int id;
	private int frequency;

	public MigratoryBird(){
		this.id = 0;
		this.frequency = 0;
	}

	public void setId(int id){
		this.id = id;
	}

	public int getId(){
		return this.id;
	}

	public int getFrequency(){
		return this.frequency;
	}

	public void restoreFrequency(){
		this.frequency = 1;
	}

	public void incrementFrequency(){
		this.frequency++;
	}

	public void copy(MigratoryBird bird){
		this.id = bird.getId();
		this.frequency = bird.getFrequency();
	}
}

File: solutions/ruby/migratory_bird.rb
class MigratoryBird

	attr_accessor :id
	attr_reader :frequency

	def initialize
		@frequency = 0
		@id = 0
	end

	def increment_frequency
		@frequency = @frequency + 1
	end

	def restore_frequency
		@frequency = 1
	end

	def copy bird
		@frequency = bird.frequency
		@id = bird.id
	end
end